pkg/destroy/gcp/dns.go

```diff
@@ -105,17 +114,19 @@ func possibleZoneParents(dnsDomain string) []string {
 	return result
 }
 
-func getParentDNSZone(dnsDomain string, publicZones []dnsZone, logger logrus.FieldLogger) *dnsZone {
+func getParentDNSZones(dnsDomain string, publicZones []dnsZone, logger logrus.FieldLogger) []*dnsZone {
+	possibleParentZones := []*dnsZone{}
+
 	possibleParents := possibleZoneParents(dnsDomain)
 	for _, parentDomain := range possibleParents {
 		for _, zone := range publicZones {
 			if zone.domain == parentDomain {
-				logger.Debugf("Found parent dns zone: %s", zone.name)
-				return &dnsZone{name: zone.name, domain: parentDomain}
+				logger.Debugf("Found possible parent dns zone: %s", zone.name)
```

Comment:
Nit: Is the `possible` qualifier here necessary? From what I understand, the zones returned are later used as if they *were* parent zones, not *maybe let's double-check*.